airbyte-integrations/connectors/source-salesforce/pyproject.toml

```diff
@@ -18,7 +18,7 @@ include = "source_salesforce"
 [tool.poetry.dependencies]
 python = "^3.9,<3.12"
 pandas = "==2.2.0"
-airbyte-cdk = "==0.59.2"
+airbyte-cdk = "^0.63.2"
```

Comment:
Tests will be failing as @brianjlai changed the way errors are being thrown in the CDK. I suggest that we leave it at 0.59.2 for the purpose of the oncall and that we fix that later on

Reply:
version 0.59.2 fails on abnormal_state test, that's why I upgraded to latest one

Reply:
@brianjlai to be clear on what the issue is here. The [previous AbstractSource code behavior](https://github.com/airbytehq/airbyte/blob/9190517649a47cc8b0b68ceadc3ad43df9ec427a/airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py#L135-L140) was:
* Catch the exception
* Do something
* Re-raise the same exception

The [new behavior](https://github.com/airbytehq/airbyte/blob/08fb4f775fc599d33a9d03d68a0a218dd9b85b85/airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py#L171-L177) is:
* Catch the exception and add it to a map <stream name, exception>
* Process other streams
* Raise AirbyteTracedException with all the streams errors

This is a breaking change as [SourceSalesforce is expecting this error specifically](https://github.com/airbytehq/airbyte/blob/08fb4f775fc599d33a9d03d68a0a218dd9b85b85/airbyte-integrations/connectors/source-salesforce/source_salesforce/source.py#L250-L251). Would it make sense that if there is only one error that we rethrow the same error? Else, we will need to do some development but the new code somewhat swallow the original error and only expose the `error_message` which seems flaky to me